### PR TITLE
feat(scripts): update install.sh to use packages on macOS

### PR DESCRIPTION
**Pull request by amdprophet**
Updates `scripts/install.sh` to install otelcol-sumo using packages when run on macOS. I forked the packaging repository to create an example release for testing. Set `PACKAGE_GITHUB_ORG="sensu"` inside `install.sh` for testing installation from a GitHub Release, or use `--binary-branch main` to test installing from GitHub Action artifacts.

These changes should be broken up into separate functions and the install script should be refactored to account for platform differences but @portertech wants to tackle that separately from this initial implementation.

### CODE REVIEW

**Review by sumo-drosiek — COMMENTED** [review]
Are `psutil` and `launchcl` available on all mac platforms?

It would be nice to have e2e tests for this PR, as it is quite significant

**Comment by amdprophet** [comment]
Yes, both plutil & launchctl exist for all macOS platforms and have been there for 10+ years. E2E tests are a great idea; I'll fix up the broken test(s) and start to add coverage today.

**Comment by portertech** [comment]
Fixed an uninstallation bug https://github.com/SumoLogic/sumologic-otel-collector-packaging/pull/10

**Comment by portertech** [comment]
@amdprophet @sumo-drosiek @perk-sumo We cannot merge this PR until the UI team is ready for the rollout across deploys. The UI team is putting a dynamic property in place to pin the MacOS installation script release (i.e. 0.79.0 instead of latest). Once this is done, we can then merge this PR and cut a collector release. The UI team can then flip a feature flag on the Staging deployment and QE can run through the E2E test suites (and make corrections etc). After E2E tests pass, the UI team can do a slow rollout across deploys (days).

@amdprophet in the meantime, please get builds passing and do further cleanup. I don't expect a green light until late next week 😓

File: pkg/scripts_test/check_darwin.go
Comment (sumo-drosiek) [inline pkg/scripts_test/check_darwin.go]: I think we may use `switch` here

File: scripts/install.sh
Comment (sumo-drosiek) [inline scripts/install.sh:726]: Do we need this change? If yes, can we linux with it?

**Review by sumo-drosiek — APPROVED** [review]
two minor things

File: scripts/install.sh
Comment (amdprophet) [reply scripts/install.sh:726]: This is necessary on macOS as `common.yaml` is replaced on upgrades. I haven't been able to find a way around this. I'll update the logic for Linux to match.

**Pull request merged**
